src/Runner.Listener/Runner.cs

```diff
@@ -37,6 +37,8 @@ public sealed class Runner : RunnerService, IRunner
         private readonly object _authMigrationTelemetryLock = new();
         private IRunnerServer _runnerServer;
         private CancellationTokenSource _authMigrationTelemetryTokenSource = new();
+        private bool _runnerExiting = false;
+        private bool _hasTerminationGracePeriod = false;
```

Comment:
i use `_hasTerminationGracePeriod` as the feature flag, so unless the new ENV is set, no new code will get executed.